File: lib/limelight/commands/pack_command.rb
#- Copyright © 2008-2010 8th Light, Inc. All Rights Reserved.
#- Limelight and all included source files are distributed under terms of the GNU LGPL.

require 'limelight/commands/command'

module Limelight
  module Commands

    # See the following usage summary.
    #
    #  Usage: limelight pack <production_path>
    #      Pack a limelight production into a .llp file.
    #      options:
    #      -h, --help                       Prints this usage summary.
    #      -n, --name                       Name of the generated llp file.  Defaults to the production folder name.
    #
    class PackCommand < Command

      install_as "pack"

      def self.description #:nodoc:
        return "Pack a limelight production into a .llp file."
      end

      protected ###########################################

      def parameter_description #:nodoc:
        return "<production_path>"
      end

      def parse_remainder(args) #:nodoc:
        @production_path = args.shift
        raise "Missing production path" if @production_path.nil?
      end

      def process #:nodoc:
        packer = Limelight::Util::Packer.new
        if @llp_name
          packer.pack(@production_path, @llp_name)
        else
          packer.pack(@production_path)
        end
      end

      def build_options(spec) #:nodoc:
        spec.on("-n <llp_name>", "--name=<llp_name>", "Name of the generated llp file.  Defaults to the production folder name.") { |value| @llp_name = value}
      end
    end

  end


end

File: src/limelight/commands/PackCommand.java
//- Copyright © 2008-2010 8th Light, Inc. All Rights Reserved.
//- Limelight and all included source files are distributed under terms of the GNU LGPL.

package limelight.commands;

import limelight.io.FileUtil;
import limelight.io.Packer;

import java.util.Map;

public class PackCommand extends Command
{
  private static Arguments arguments;
  private Packer packer;

  public static Arguments arguments()
  {
    if(arguments == null)
    {
      arguments = new Arguments();
      arguments.addParameter("production", "Path to production folder to be packed");
      arguments.addValueOption("n", "name", "llp-name", "Name of the generated llp file.  Defaults to the production folder name.");
    }
    return arguments;
  }

  @Override
  public void doExecute(Map<String, String> args)
  {
    String productionDir = args.get("production");
    String llpName = args.get("name");
    if(llpName != null)
      getPacker().pack(productionDir, llpName);
    else
      getPacker().pack(productionDir, FileUtil.baseName(productionDir));
  }

  private Packer getPacker()
  {
    if(packer == null)
      packer = new Packer();
    return packer;
  }

  public void setPacker(Packer packer)
  {
    this.packer = packer;
  }

  @Override
  public String description()
  {
    return "Pack a limelight production into a .llp file.";
  }

  @Override
  public String name()
  {
    return "pack";
  }

  @Override
  public Arguments getArguments()
  {
    return arguments();
  }
}
